executors/executor.go

```diff
@@ -82,6 +87,66 @@ type VolumeRequest struct {
 	Arbiter bool
 }
 
+type VolumeCloneRequest struct {
+	Volume string
+	Clone  string
+}
+
+type VolumeSnapshotRequest struct {
+	Volume      string
+	Snapshot    string
+	Description string
+}
+
+type SnapshotCloneRequest struct {
+	Volume   string
+	Snapshot string
+}
+
+type Snapshot struct {
+	XMLName xml.Name `xml:"snapshot"`
```

Comment:
I'm a bit puzzled as to why we have xml.Name types in what I feel is the interface to the "outside world." To me it seems like use of xml is an implementation detail. If these are not meant to be part of the public api, perhaps they could be moved to a different file?

Reply:
Maybe that would have been nicer, but there are other structs in the same file that already do it like this.

Reply:
OK, I missed that there was a precedent for this already in that file. Cleanups can always come later.

Reply:
Yes, blame shows it was me who introduced it and I don't seem to recall why I named it so. I don't like it either. We can have cleanup patch for this.

Reply:
Yeah, let's cleanup later.